### PR TITLE
Add shard assignment in validator RPC; fix main.go for staking validator setup

### PR DESCRIPTION
**Pull request by rlan35**
tested locally and with devnet.

### CODE REVIEW

File: cmd/harmony/main.go [hunk 1]
```diff
@@ -209,6 +209,7 @@ func setupInitialAccount() (isLeader bool) {
 	}
 
 	if initialAccount == nil {
+		initialAccount = &genesis.DeployAccount{}
```
Comment (harmony-ek) [inline cmd/harmony/main.go]: \*facepalm\* LOL

File: hmy/api_backend.go [hunk 2]
```diff
@@ -362,3 +362,12 @@ func (b *APIBackend) GetValidatorSelfDelegation(addr common.Address) *big.Int {
 	}
 	return wrapper.Delegations[0].Amount
 }
+
+// GetShardState ...
+func (b *APIBackend) GetShardState() *shard.State {
```
Comment (harmony-ek) [inline hmy/api_backend.go]: `shard.State` is a slice (defined as `[]shard.Committee`), so please just return the slice itself, not a pointer to it.  Also please return an error code:
```suggestion
func (b *APIBackend) GetShardState() (shard.State, error) {
```

File: hmy/api_backend.go [hunk 3]
```diff
@@ -362,3 +362,12 @@ func (b *APIBackend) GetValidatorSelfDelegation(addr common.Address) *big.Int {
...
+	state, err := b.hmy.BlockChain().ReadShardState(b.hmy.BlockChain().CurrentHeader().Epoch())
+	if err != nil {
+		return &shard.State{}
```
Comment (harmony-ek) [inline hmy/api_backend.go]:
```suggestion
		return nil, errors.Wrapf(err, "cannot read from DB")
```

File: hmy/api_backend.go [hunk 4]
```diff
@@ -362,3 +362,12 @@ func (b *APIBackend) GetValidatorSelfDelegation(addr common.Address) *big.Int {
...
+	}
+	return &state
```
Comment (harmony-ek) [inline hmy/api_backend.go]:
```suggestion
	return state, nil
```

File: internal/hmyapi/backend.go [hunk 5]
```diff
@@ -80,6 +80,7 @@ type Backend interface {
 	GetDelegationsByValidator(validator common.Address) []*staking.Delegation
 	GetDelegationsByDelegator(delegator common.Address) ([]common.Address, []*staking.Delegation)
 	GetValidatorSelfDelegation(addr common.Address) *big.Int
+	GetShardState() *shard.State
```
Comment (harmony-ek) [inline internal/hmyapi/backend.go]:
```suggestion
	GetShardState() (shard.State, error)
```

File: internal/hmyapi/blockchain.go [hunk 6]
```diff
@@ -518,6 +518,26 @@ func (s *PublicBlockChainAPI) GetValidatorInformation(ctx context.Context, addre
 		rpcValidator.TotalEffectiveStake = stats.TotalEffectiveStake.String()
 	}
 
+	shardState := s.b.GetShardState()
```
Comment (harmony-ek) [inline internal/hmyapi/blockchain.go]:
```suggestion
	shardState, err := s.b.GetShardState()
```

File: internal/hmyapi/blockchain.go [hunk 7]
```diff
@@ -518,6 +518,26 @@ func (s *PublicBlockChainAPI) GetValidatorInformation(ctx context.Context, addre
...
+	if shardState != nil {
```
Comment (harmony-ek) [inline internal/hmyapi/blockchain.go]:
```suggestion
	if err == nil {
```

File: internal/hmyapi/blockchain.go [hunk 8]
```diff
@@ -518,6 +518,26 @@ func (s *PublicBlockChainAPI) GetValidatorInformation(ctx context.Context, addre
...
+		blsKeyToShardID := make(map[shard.BlsPublicKey]uint32)
+
+		for _, committee := range *shardState {
```
Comment (harmony-ek) [inline internal/hmyapi/blockchain.go]:
```suggestion
		for _, committee := range shardState {
```

File: cmd/staking/root.go [hunk 9]
```diff
@@ -121,13 +121,13 @@ func (s *staker) run(cmd *cobra.Command, args []string) error {
 			return staking.DirectiveDelegate, staking.Delegate{
 				dAddr,
 				dAddr2,
-				big.NewInt(1000),
+				big.NewInt(0).Mul(big.NewInt(denominations.One), big.NewInt(10)),
```
Comment (harmony-ek) [inline cmd/staking/root.go]: What does this achieve?

File: cmd/harmony/main.go [hunk 10]
```diff
@@ -463,7 +464,7 @@ func main() {
 		setupInitialAccount()
 	}
 
-	if *shardID >= 0 {
```
Comment (harmony-ek) [inline cmd/harmony/main.go]: Is this simply a safety check against accidental override of shard ID?

Comment (harmony-ek) [reply cmd/harmony/main.go]: nvm, I read it backward.  It actually allows manual override of shard ID.

File: cmd/staking/root.go (hunk 9 above)
Comment (rlan35) [reply cmd/staking/root.go]: Nothing, just testing change. to be able to delegate more tokens.

**Comment by harmony-ek** [comment]
Jenkins, test this please.

**Pull request merged**